packages/blade/src/components/Badge/Badge.tsx

```diff
@@ -0,0 +1,76 @@
+import type { ReactElement } from 'react';
+import type { CSSObject } from 'styled-components';
+import styled from 'styled-components';
+import { StyledBadge } from './StyledBadge';
+import type { IconComponent } from '~components/Icons';
+import { BaseText } from '~components/Typography/BaseText';
+import { getPlatformType } from '~utils';
+import Box from '~components/Box';
+
+type BadgeProps = {
+  children: string;
+  variant?: 'positive' | 'negative' | 'notice' | 'information' | 'neutral';
+  contrast?: 'low' | 'high';
+  size?: 'small' | 'medium';
+  icon?: IconComponent;
+  fontWeight?: 'regular' | 'bold';
+};
+
+const StyledBaseText = styled(BaseText)(
+  (): CSSObject => {
+    if (getPlatformType() !== 'react-native') {
+      return {
+        whiteSpace: 'nowrap',
+        overflow: 'hidden',
+        textOverflow: 'ellipsis',
+      };
+    }
+    return {};
+  },
+);
+
+const Badge = ({
+  children,
+  contrast = 'low',
+  fontWeight = 'regular',
+  icon: Icon,
+  size = 'medium',
+  variant = 'neutral',
+}: BadgeProps): ReactElement => {
+  if (!children?.trim()) {
+    throw new Error('[Blade: Badge]: Text as children is required for Badge.');
+  }
+
+  return (
+    <StyledBadge variant={variant} contrast={contrast} size={size}>
+      <Box
+        paddingRight="spacing.2"
+        paddingLeft="spacing.2"
+        display="flex"
+        flex={1}
```

Comment:
Is this needed?

Reply:
@chaitanyadeorukhkar withIcon story isn't working as intended

<img width="437" alt="image" src="https://user-images.githubusercontent.com/35374649/183901678-2b4fceef-badf-45bb-9014-cedfad467f2b.png">

Reply:
Can you check the CI link generated for chromatic? Local messes up sometimes. It looks fine on my local

Reply:
Yeah that's weird. Working on chromatic. Why is this happening?

Reply:
Oh i think storybook is messing up because in argsType we are defining the control as `select` with option with `Object.keys`. 

But then in `BadgesWithVariantTemplate` we are not doing: 
```ts
const IconComponent = iconMap[(args.icon as unknown) as string];
```

Reply:
Generally I clear storybook's cache and everything works okay